src/State/BoardCreator/SaveBoardCreatorState.cpp

```diff
@@ -0,0 +1,45 @@
+#include <State/BoardCreator/SaveBoardCreatorState.hpp>
+#include <Graphics/GfxStream.hpp>
+#include <State/BoardCreator/MoveBoardCreatorState.hpp>
+#include <iomanip>
+
+const int SaveBoardCreatorState::MAX_FILENAME_LENGTH = 20;
+
+SaveBoardCreatorState::SaveBoardCreatorState(int size) : AbstractBoardCreatorState(size) {
+
+}
+
+SaveBoardCreatorState::SaveBoardCreatorState(std::shared_ptr<Board> board) :
+        AbstractBoardCreatorState(std::move(board)) {
+
+}
+
+SaveBoardCreatorState::SaveBoardCreatorState(std::shared_ptr<BoardCreator> boardCreator) :
+        AbstractBoardCreatorState(std::move(boardCreator)) {
+
+}
+
+void SaveBoardCreatorState::HandleInput(Game &game, char input) {
+    if (isalnum(input) && filename.length() < MAX_FILENAME_LENGTH) {
+        filename += input;
+    } else if (filename.length() > 0 && (input == 127 || input == 8)) { // backspace
+        filename.pop_back();
+    } else if (input == '\n') {
+        boardCreator->GetBoard()->SaveToFile(filename);
+        game.SetState(std::make_shared<MoveBoardCreatorState>(std::move(boardCreator)));
+    } else if (input == 27) { // ESC
```

Comment:
może lepiej jakoś statycznie to określić albo enumować, bo to liczba z dupy w kodzie :D